platform/iphone/gl_view.mm

```diff
@@ -379,369 +222,106 @@ - (void)destroyFramebuffer {
+		// This is an extremely terrible way to animate. Very low resolution with lots
+		// of drift. You would be lucky to have this called every 16ms
+		animationTimer = [NSTimer scheduledTimerWithTimeInterval:self.animationInterval
+														  target:self
+														selector:@selector(drawView)
+														userInfo:nil
+														 repeats:YES];
 	}
 
-	if (video_playing) {
-		_unpause_video();
-	}
+	// if (video_playing) {
+	// 	_unpause_video();
+	// }
 }
 
 - (void)stopAnimation {
-	if (!active)
-		return;
-	active = FALSE;
-	printf("******** stop animation!\n");
+	if (!self.isActive) return;
+
+	self.active = NO;
+	printf("stop animation!\n");
 
-	if (useCADisplayLink) {
-		[displayLink invalidate];
-		displayLink = nil;
+	if (self.useCADisplayLink) {
+		[_displayLink invalidate];
+		_displayLink = nil;
 	} else {
 		[animationTimer invalidate];
 		animationTimer = nil;
 	}
 
-	clear_touches();
-
-	if (video_playing) {
-		// save position
-	}
+	// if (video_playing) {
+	// 	_pause_video();
+	// }
 }
 
 - (void)setAnimationInterval:(NSTimeInterval)interval {
-	animationInterval = interval;
-	if ((useCADisplayLink && displayLink) || (!useCADisplayLink && animationTimer)) {
+
+	_animationInterval = interval;
+
+	if (self.isActive) {
 		[self stopAnimation];
-		[self startAnimation];
 	}
+	[self startAnimation];
 }
 
-// Updates the OpenGL view when the timer fires
-- (void)drawView {
-	if (useCADisplayLink) {
-		// Pause the CADisplayLink to avoid recursion
-		[displayLink setPaused:YES];
-
-		// Process all input events
-		while (CFRunLoopRunInMode(kCFRunLoopDefaultMode, 0, TRUE) == kCFRunLoopRunHandledSource)
-			;
-
-		// We are good to go, resume the CADisplayLink
-		[displayLink setPaused:NO];
-	}
+- (NSTimeInterval)animationInterval {
+	return _animationInterval;
+}
 
-	if (!active) {
-		printf("draw view not active!\n");
+- (void)drawView {
+	if (!self.isActive) {
+		printf("Attempted to drawView while inactive!\n");
 		return;
-	};
+	}
 
 	// Make sure that you are drawing to the current context
 	[EAGLContext setCurrentContext:context];
 
 	// If our drawing delegate needs to have the view setup, then call -setupView: and flag that it won't need to be called again.
-	if (!delegateSetup) {
-		[delegate setupView:self];
-		delegateSetup = YES;
+	if (!self.isSetUpComplete && [self.delegate respondsToSelector:@selector(setupView:)]) {
+		[self.delegate setupView:self];
+		self.setUpComplete = YES;
 	}
 
 	glBindFramebufferOES(GL_FRAMEBUFFER_OES, viewFramebuffer);
-
-	[delegate drawView:self];
+	[self.delegate drawView:self];
 
 	glBindRenderbufferOES(GL_RENDERBUFFER_OES, viewRenderbuffer);
 	[context presentRenderbuffer:GL_RENDERBUFFER_OES];
 
 #ifdef DEBUG_ENABLED
 	GLenum err = glGetError();
 	if (err)
-		NSLog(@"%x error", err);
+		NSLog(@"%x (gl) error", err);
 #endif
 }
 
-- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event {
-	NSArray *tlist = [[event allTouches] allObjects];
```

Comment:
These touch handlers took me a bit of understanding. Basically when any of these handlers is called by iOS `touches` will contain a `UITouch` for each finger currently on the screen. There's no need to iterate over or validate this list. The fact that iOS has called the specific `touches...` method is validation enough. 

Additionally, if you're in `touchesBegan` you'll *never* receive a touch with a `phase` other than `UITouchPhaseBegan`, there's simply no reason to check that.

Finally, it seems like the `get_touch_id()` function and friends existed purely to provide a constant `int` index back to Godot as each touch passes through each phase. We've chosen instead to store the touches in an array and pass the index around.